t/rose-task-run/31-app-bunch.t

```diff
@@ -163,5 +163,59 @@ for KEY in $(seq 0 3); do
         "ROSE_BUNCH_LOG_PREFIX: $KEY" $FILE_DIR/bunch.$KEY.out
 done
 #-------------------------------------------------------------------------------
+#-------------------------------------------------------------------------------
+# Testing ROSE_BUNCH_ARGUMENT_MODE_IZIP shortens all bunch-args
+#-------------------------------------------------------------------------------
+APP=bunch_argument_mode_izip
+#-------------------------------------------------------------------------------
+TEST_KEY_PREFIX=argument_mode_izip
+FILE=$LOG_DIR/$APP/01/job.out
+ARG1=1
+ARG2=a
+ARG3=9
+file_grep "$TEST_KEY_PREFIX-RUN-$ARG1-$ARG2-$ARG3" \
+    "\[INFO\] [0-9]*-[0-9]*-[0-9]*T[0-9]*:[0-9]*:[0-9]*+[0:9]* echo arg1: $ARG1 - arg2: $ARG2 - arg3: $ARG3"\
+    "$FILE"
+file_grep "$TEST_KEY_PREFIX-TOTAL-RAN" \
+    "\[INFO\] [0-9]*-[0-9]*-[0-9]*T[0-9]*:[0-9]*:[0-9]*+[0:9]* TOTAL: 1" \
+    "$FILE"
+#-------------------------------------------------------------------------------
+#-------------------------------------------------------------------------------
+# Testing ROSE_BUNCH_ARGUMENT_MODE_IZIP_LONGEST lengthens all bunch-args
+#-------------------------------------------------------------------------------
+APP=bunch_argument_mode_izip_longest
+#-------------------------------------------------------------------------------
+TEST_KEY_PREFIX=argument_mode_izip_longest
+FILE=$LOG_DIR/$APP/01/job.out
+for VALUES in 1,a,9 2,,23 3,, 4,,; do
+    IFS=, read -r -a ARGS <<< "$VALUES"
+    file_grep "$TEST_KEY_PREFIX-RUN-$ARG1-$ARG2-$ARG3" \
```

Comment:
I had to investigate this test, since all of the sub-tests were named identically, suggesting there might be an issue. Firstly, the arguments to the test name are those from the ``izip``-related test instead of the ones tested:

```suggestion
    file_grep "$TEST_KEY_PREFIX-RUN-${ARGS[0]:-}-${ARGS[1]:-}-${ARGS[2]:-}" \
```

However, besides this I do get the impression that these ``izip``-related tests do not test the intended behaviour; when I arbitrarily change the values provided to the ``VALUES`` in the ``for`` loop, the tests still pass. Could you please take another look at this & confirm or otherwise if this does as it should?

As a general comment, I am struggling to follow the intended behaviour for each sub-test for the two ``izip-*`` argument modes. If you had the time, it would be useful if you could add some brief comments to clarify the expected test outputs (for all three modes).